### ISSUE TITLE
Apply `--safe` and `--without-k` on a library level

### ISSUE THREAD

**Issue by MatthewDaggitt** [agda/agda-stdlib#1524]
Now that Agda 2.6.2 has been released, we don't need the `OPTIONS` pragmas everywhere, we can instead specify the default options in the `standard-library.agda` file. This should zap several hundred lines of code in the process :tada:

**Comment by gallais** [comment]
Surely local flags cannot invalidate the ones set globally?

**Comment by jespercockx** [comment]
Yes they can, see https://github.com/agda/agda/issues/5265#issuecomment-792256883

**Comment by gallais** [comment]
So this means that we cannot trust a `--safe` in the lib file. This is distressing.

**Comment by jespercockx** [comment]
I think `--safe` currently has a special status that means it is treated differently from other flags. In particular there is no way to *disable* `--safe` explicitly so it's not possible to turn it off once it has been turned on explicitly (be it in the file, on the command line, or in the agda-lib file). However, for other flags such as `--without-K` this is currently not the case, which is what that issue is about.

**Comment by MatthewDaggitt** [comment]
Hmm, so sounds like we need two different fields in the library file `default-flags` which can be overridden and `always-flags` which can't? I'll open an Agda issue to discuss.
